src/metabase/sync/sync_metadata/sync_table_privileges.clj

```diff
@@ -0,0 +1,36 @@
+(ns metabase.sync.sync-metadata.sync-table-privileges
+  (:require
+   [clojure.java.jdbc :as jdbc]
+   [honey.sql :as sql]
+   [metabase.driver :as driver]
+   [metabase.driver.util :as driver.u]
+   [metabase.models.interface :as mi]
+   [metabase.util.malli :as mu]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(mu/defn sync-table-privileges!
+  "Sync the `table_privileges` table with the privileges in the database.
+
+   This is a cache of the data returned from `driver/table-privileges`, but it's stored in the database for performance."
+  [database :- (mi/InstanceOf :model/Database)]
+  (let [driver (driver.u/database->driver database)]
+    ;; The following expression should be replaced by (when (driver/supports? :actions driver) ...)
+    ;; when table_privileges are implemented for all the actions drivers
+    (when (contains? (methods driver/current-user-table-privileges) driver)
+      (let [rows               (driver/current-user-table-privileges driver database)
+            schema+table->id   (t2/select-fn->pk (fn [t] {:schema (:schema t), :table (:name t)}) 'Table :db_id (:id database))
```

Comment:
```suggestion
            schema+table->id   (t2/select-fn->pk (fn [t] {:schema (:schema t), :table (:name t)}) :model/Table :db_id (:id database))
```

Using `'Table` comes with a small performance cost because it needs to look up the model.